src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageSourceMappingActionViewModel.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class PackageSourceMappingActionViewModel : ViewModelBase
+    {
+        internal INuGetUI UIController { get; }
+
+        private PackageSourceMappingActionViewModel(INuGetUI uiController)
+        {
+            UIController = uiController ?? throw new ArgumentNullException(nameof(uiController));
+        }
+
+        public bool IsPackageSourceMappingEnabled => UIController.UIContext?.PackageSourceMapping?.IsEnabled == true;
+
+        private string? _packageId;
+
+        internal string? PackageId
+        {
+            get { return _packageId; }
+            set
+            {
+                if (_packageId == value)
+                {
+                    return;
+                }
+
+                _packageId = value;
+
+                RaisePropertyChanged(nameof(IsPackageMapped));
+                RaisePropertyChanged(nameof(MappingStatus));
```

Comment:
The ViewModelBase class you created is great! The pattern of checking if the value changed and only then updating and raising property changed is a very common one. You can consider adding to that base class is a method that takes in a variable reference and the value and determines if the value changed, and if so, updates the variable reference to the `value` and raises a propery changed on the property (by the caller's name) and returns a boolean indicating if the property was updated or not so we can use the boolean to do something like this in the caller:

```
if (SetAndRaisePropertyChanged(ref _packageId))
{
  // IsPackageMapped will already be raised since it's the name of the caller member name
  // but now that we know the value changed, we also want to raise any property changed
  // for other properties that are affected by this update
  RaisePropertyChanged(nameof(MappingStatus));
}
```

Reply:
Yes, I do have that method in another branch where I need such a method. However, I didn't implement it now since nothing needs to SetAndRaise. For example, nothing listens to PackageID, so I do not want to take the suggestion to raise property changed on that property in this PR.